lib/data/external/AwsClient.js

```diff
@@ -62,6 +62,7 @@ class AwsClient {
         const stream = request.createReadStream().on('error', err => {
             log.error('error streaming data from AWS', { error: err,
                 dataStoreName: this._dataStoreName });
+            return callback(err);
```

Comment:
why is this added? didn't we discuss that this callback is confusing since it is not actually called?  once the listener is put on the stream, the headers have already been sent back to the requesting client and we break the connection if there is an error.